src/librustc_mir/transform/add_yk_swt_calls.rs

```diff
@@ -18,8 +18,28 @@ use rustc::hir;
 use rustc::hir::def_id::{DefIndex, LOCAL_CRATE};
 use rustc::hir::map::blocks::FnLikeNode;
 
-/// A MIR pass which, for each basic block, inserts a call to the software trace recorder.
-/// The call arguments passed uniquely identify the MIR location.
+/// A MIR transformation that, for each basic block, inserts a call to the software trace recorder.
+/// The arguments to the calls (crate hash, DefId and block index) identify the position to be
+/// inserted into a trace.
+///
+/// The transformation works by copying each block and replacing the original with a new block
+/// which calls the trace recorder function. The call then returns to the copy.
+///
+/// For example:
+///
+/// +----+   +----+           +-----+   +-----+   +----+   +-----+   +-----+   +----+
+/// | B0 |-->| B1 |  Becomes: | B0' |-->| Rec |-->| B2 |-->| B1' |-->| Rec |-->| B3 |
+/// +----+   +----+           +-----+   +-----+   +----+   +-----+   +-----+   +----+
+///
+/// Where:
```

Comment:
> The structure of the CFG makes the ordering of the blocks completely clear doesn't it?

In the ASCII art diagram, the blocks are not shown in order they appear in the vector, which is why I named the blocks with their indices.

> On the matter of the name, maybe we could call the new blocks "record blocks"?

I'm not keen on "record blocks", because IDs that are recorded into the trace are in fact the shadows. How about `copy_blks`?

Reply:
After a quick conversation we're thinking that "shadow blocks" and "user blocks" might be appropriate terms.